### ISSUE TITLE
`wp_function_not_compatible_with_requires_wp` false positive when call is guarded by `function_exists()`

### ISSUE THREAD

**Issue by marekdedic** [10up/plugin-check#1363]
### Description

`WP_Functions_Compatibility_Check` reports `wp_function_not_compatible_with_requires_wp` on a function call that is already guarded by `function_exists()`, making the finding a false positive — the plugin already handles the incompatibility at runtime.

### Steps to reproduce

1. Declare `Requires at least: 4.9.6` in the plugin header.
2. Write code that conditionally registers a block only when the block API is available:

```php
public function __construct() {
    if ( function_exists( 'register_block_type' ) ) {
        add_action( 'init', array( self::class, 'add' ) );
    }
}

public static function add() {
    register_block_type( 'my-plugin/gallery', array( /* ... */ ) );
}
```

3. Run `wp plugin check my-plugin`.

### Actual behavior

```
FILE: frontend/class-block.php
line  column  type   code                                      message
95    0       ERROR  wp_function_not_compatible_with_requires_wp  Function "register_block_type()" requires WordPress 5.0.0, but your plugin minimum supported version is WordPress 4.9.6.
```

### Expected behavior

No error. The plugin already guards against the incompatibility with `function_exists('register_block_type')`, so `add()` is never called on WordPress versions that lack the function. The checker should suppress the finding when the flagged call is reachable only through a `function_exists()` guard for the same function name.

### Notes

`WP_Functions_Compatibility_Check` uses PHP tokenization (`token_get_all`) to find global function calls, but performs no control-flow analysis. It cannot currently see that a call site is conditionally protected. Recognising `if ( function_exists( 'foo' ) ) { ... foo() ... }` as a safe pattern would eliminate this class of false positive.

This false positive is not coming from a PHPCS rule, so it cannot be disabled in a targeted way, only by disabling the check altogether.

*Disclaimer: Issue was analyzed and written using Claude, but the output was verified by me.*

**Comment by shameemreza** [comment]
Thanks for the report, confirmed as a false positive. I reproduced it with a minimal plugin, and the check flags the guarded call.

Plugin header set to `Requires at least: 4.9.6`, same structure as your report:

```
class PCP_Repro_Block {
    public function __construct() {
        if ( function_exists( 'register_block_type' ) ) {
            add_action( 'init', array( self::class, 'add' ) );
        }
    }
    public static function add() {
        register_block_type( 'my-plugin/gallery', array( 'render_callback' => '__return_empty_string' ) );
    }
}
```

Running the check:

```
FILE: frontend/class-block.php
line  type   code                                         message
18    ERROR  wp_function_not_compatible_with_requires_wp  Function "register_block_type()" requires WordPress 5.0.0, but your plugin minimum supported version is WordPress 4.9.6.
```

The cause is in `WP_Functions_Compatibility_Check`. It collects calls with `token_get_all()` and matches any `T_STRING` followed by `(`, with no control flow analysis, so it never sees the `function_exists()` guard.

One thing worth calling out about your example. The guard sits in `__construct()` and the call sits in `add()`, wired together through `add_action`. A narrow "guard wraps the call" rule wouldn't catch it, since the two live in different methods. A same scope check would still report your case.

The plan is a file level heuristic. If `function_exists( 'foo' )` appears anywhere in the same file, suppress the compatibility finding for `foo()` in that file. That covers the deferred hook pattern you hit while staying within what the tokenizer can do. The trade off is a rare false negative if someone guards in one place but also calls the function unconditionally elsewhere, which is acceptable since `function_exists()` is the documented pattern for this.

CC @davidperezgar

**Comment by davidperezgar** [comment]
That's a good catch!
